misaligned.cpp

```diff
@@ -1,21 +1,39 @@
 #include <iostream>
 #include <assert.h>
 
+void checkPairNumberValidity(int pair_no)
+{
+    assert(pair_no>=1 && pair_no<=25);
+}
+
+void displayPairNumber(int x, int y)
+{
+    int pair_number = x * 5 + y;
+    std::cout<<pair_number+1;   //Pair numbers should start from 1
+    checkPairNumberValidity(pair_number);
+}
+
+
+void displayColourPair(int major, int minor)
+{
+    const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+    const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+    std::cout << "|" << majorColor[major] <<"|"<<minorColor[minor]<<"\n";
+}
+
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
+    int i , j = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            displayPairNumber(i, j);
+            displayColourPair(i, j);
         }
     }
     return i * j;
 }
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
+    printColorMap();

```

Comment:
how is the output verified? there needs to be a way to verify whatever is printed or atleast performed by the function.
One way would be to keep a list of strings which is nothing but what is printed to console and then check that list of correctness.